File: app/src/main/java/woo/sopt22/meowbox/View/Home/CardViewAdapter.java
package woo.sopt22.meowbox.View.Home;

import android.content.Context;
import android.content.Intent;
import android.media.Image;
import android.support.annotation.NonNull;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;

import java.util.List;

import woo.sopt22.meowbox.R;
import woo.sopt22.meowbox.Util.ToastMaker;
import woo.sopt22.meowbox.View.MeowBoxDetail.MeowBoxDetailActivity;

public class CardViewAdapter extends PagerAdapter{
    private LayoutInflater inflater;
    private List<CardData> mPageList;
    private Context context;
    private View.OnClickListener onItemClick;

    public CardViewAdapter(LayoutInflater inflater, List<CardData> mPageList) {
        this.inflater = inflater;
        this.mPageList = mPageList;
    }

    @Override
    public int getCount() {
        return mPageList.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        View view = inflater.inflate(R.layout.cardview_adapter, container, false);


        ImageView home_detail_btn = (ImageView) view.findViewById(R.id.home_detail_btn);
        home_detail_btn.setOnClickListener(onItemClick);
        ImageView mImageView = view.findViewById(R.id.titleImageView);
        CardData item = mPageList.get(position);
        Glide.with(container.getContext()).load(item.getImage()).into(mImageView);
        container.addView(view);
        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((View) object);
    }


    public void setOnItemClickListener(View.OnClickListener l){
        onItemClick = l;
    }
}

File: app/src/main/java/woo/sopt22/meowbox/View/Home/CardViewAdapter.kt
package woo.sopt22.meowbox.View.Home

import android.content.Context
import android.content.Intent
import android.media.Image
import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView

import com.bumptech.glide.Glide

import woo.sopt22.meowbox.R
import woo.sopt22.meowbox.Util.ToastMaker
import woo.sopt22.meowbox.View.MeowBoxBirthDay.MeowBoxtBirthDayStoryActivity
import woo.sopt22.meowbox.View.MeowBoxDetail.MeowBoxDetailActivity

class CardViewAdapter(private val inflater: LayoutInflater, private val mPageList: List<CardData>) : PagerAdapter() {
    private val context: Context? = null
    private var onItemClick: View.OnClickListener? = null

    override fun getCount(): Int {
        return mPageList.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = inflater.inflate(R.layout.cardview_adapter, container, false)


        val home_detail_btn = view.findViewById<View>(R.id.home_detail_btn) as ImageView
        home_detail_btn.setOnClickListener(onItemClick)
        val mImageView = view.findViewById<ImageView>(R.id.titleImageView)
        val item = mPageList[position]
        Glide.with(container.context).load(item.image).into(mImageView)
        if (item.tmp == 1) {
            home_detail_btn.setImageResource(R.drawable.home_detail_btn_white)
        } else if (item.tmp == 2) {
            home_detail_btn.visibility = View.INVISIBLE
        }
        container.addView(view)
        return view
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as View)
    }


    fun setOnItemClickListener(l: View.OnClickListener) {
        onItemClick = l
    }
}
